homeassistant/components/switcher_kis/switch.py

```diff
@@ -0,0 +1,179 @@
+    def current_power_w(self) -> int:
+        """Return the current power usage in W."""
+        return cast(int, self._device_data.power_consumption)
+
+    @property
+    def device_state_attributes(self) -> Dict:
+        """Return the optional state attributes."""
+        from aioswitcher.consts import WAITING_TEXT
+
+        attribs = {}
+
+        for prop, attr in PROPERTIES_TO_DEVICE_ATTRIBUTES.items():
+            value = getattr(self, prop)
+            if value and value is not WAITING_TEXT:
+                attribs[attr] = value
+
+        return attribs
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        from aioswitcher.consts import (STATE_OFF as SWITCHER_STATE_OFF,
+                                        STATE_ON as SWITCHER_STATE_ON)
+        return self._state in [SWITCHER_STATE_ON, SWITCHER_STATE_OFF]
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        if not self.hass:
+            return
+        self._unsub_dispatcher = async_dispatcher_connect(
+            self.hass, SIGNAL_SWITCHER_DEVICE_UPDATE, self.async_update_data)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Run when entity will be removed from hass."""
+        if self._unsub_dispatcher:
+            self._unsub_dispatcher()
+
+    async def async_update_data(self, device_data: Any) -> None:
+        """Update the entity data."""
+        # pylint: disable=unused-import
+        from aioswitcher.devices import SwitcherV2Device
+        # pylint: enable=unused-import
+        temp_device = device_data  # type: SwitcherV2Device
+        if temp_device:
+            if self._self_initiated:
+                self._self_initiated = False
+            else:
+                self._device_data = temp_device
+                self._state = self._device_data.state
+                self.async_schedule_update_ha_state()
+
+    async def async_turn_on(self, **kwargs: Dict) -> None:
+        """Turn the entity on.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        await self._control_device(True)
+
+    async def async_turn_off(self, **kwargs: Dict) -> None:
+        """Turn the entity off.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        await self._control_device(False)
+
+    async def _control_device(self, send_on: bool) -> None:
+        """Turn the entity on or off."""
+        from aioswitcher.api import SwitcherV2Api
+        # pylint: disable=unused-import
+        from aioswitcher.api.messages import SwitcherV2ControlResponseMSG
+        # pylint: enable=unused-import
+        from aioswitcher.consts import (COMMAND_OFF, COMMAND_ON,
+                                        STATE_OFF as SWITCHER_STATE_OFF,
+                                        STATE_ON as SWITCHER_STATE_ON)
+        if not self.hass:
```

Comment:
This can never happen as the only entry points are from service calls.  Remove.

Reply:
Fixed in this [commit](https://github.com/home-assistant/home-assistant/pull/22325/commits/aa9b0400a80601fa103ff3f62cccac74d397641f).